{\rtf1\ansi\ansicpg1252\cocoartf1038\cocoasubrtf360
{\fonttbl\f0\fswiss\fcharset0 Helvetica;}
{\colortbl;\red255\green255\blue255;}
\margl1440\margr1440\vieww9000\viewh8400\viewkind0
\pard\tx720\tx1440\tx2160\tx2880\tx3600\tx4320\tx5040\tx5760\tx6480\tx7200\tx7920\tx8640\ql\qnatural\pardirnatural

\f0\fs24 \cf0 diff -bru wxWidgets.orig/samples/console/console.cpp wxWidgets/samples/console/console.cpp\
--- wxWidgets.orig/samples/console/console.cpp	2014-07-31 21:15:07.000000000 -0700\
+++ wxWidgets/samples/console/console.cpp	2014-08-07 13:04:54.000000000 -0700\
@@ -30,6 +30,8 @@\
 #endif\
 \
 #include <wx/app.h>\
+#include <wx/protocol/ftp.h>\
+#include <wx/wfstream.h>\
 #include <wx/cmdline.h>\
 \
 // ============================================================================\
@@ -56,7 +58,65 @@\
         fprintf(stderr, "Failed to initialize the wxWidgets library, aborting.");\
         return -1;\
     \}\
-\
+    wxFTP ftp;\
+	if( !ftp.Connect( "ftp.wxwidgets.org" ) )\
+	\{\
+		fprintf( stderr, "Couldn't connect!" );\
+		return -1;\
+	\}\
+	bool res;\
+	printf( "Changing the directory\\n\\r" );\
+	res = ftp.ChDir( "/pub/2.8.9" );\
+	if( !res )\
+	\{\
+		fprintf( stderr, "Can't change the directory\\n\\r" );\
+		return -1;\
+	\}\
+	ftp.SetBinary();\
+	const char *filename = "wxWidgets-2.8.9.tar.bz2";\
+	int size = ftp.GetFileSize( filename );\
+	if( size == -1 )\
+	\{\
+		fprintf( stderr, "Can't get the file size" );\
+		return -1;\
+	\}\
+	wxInputStream *in = ftp.GetInputStream( filename );\
+	if( !in )\
+	\{\
+		fprintf( stderr, "Can't get the stream for downloading" );\
+		return -1;\
+	\}\
+	else\
+	\{\
+		char *buf = new char[2048];\
+		int sizeRead = 0;\
+		printf( "Starting the download...\\n\\r" );\
+		wxOutputStream *out = new wxFileOutputStream( "temp" );\
+		while( !in->Eof() )\
+		\{\
+			in->Read( buf, 2048 );\
+			out->Write( buf, 2048 );\
+			sizeRead += 2048;\
+			if( sizeRead > size / 2 )\
+			\{\
+				res = ftp.Abort();\
+				if( !res )\
+					fprintf( stderr, "Aborting error\\n\\r" );\
+				else\
+				\{\
+					printf( "Download aborted!\\n\\r" );\
+					break;\
+				\}\
+			\}\
+		\}\
+		delete[] buf;\
+		res = ftp.ChDir( "/pub/2.8.8" );\
+		if( !res )\
+		\{\
+			fprintf( stderr, "Can't change the directory\\n\\r" );\
+			return -1;\
+		\}\
+	\}\
     wxCmdLineParser parser(cmdLineDesc, argc, argv);\
     switch ( parser.Parse() )\
     \{\
}